### ISSUE TITLE
Crash on startup. java.sql.SQLException: User not found: SA

### ISSUE THREAD

**Issue by lord-carlos** [airsonic/airsonic#874]
## Problem description

*Hello. I'm using the docker from linuxservers. I don't know when the problem started, but at some point the airsonic just kept restarting*

### Steps to reproduce

*I can **not** reproduce it with a clean db. If I delete the config/db/ folder then airsonic starts up.*

1. *start airsonic docker*
2. *watch container using more and more ram, up to ~900mb*
3. *crash and it all starts again.* `java.sql.SQLException: User not found: SA`

## System information

 * **Airsonic version**: *10.1.2-RELEASE – July 28, 2018*
 * **Operating system**: *Debian stable*
 * **Proxy server**: *Nginx*

## Additional notes

TL;DR: 
`liquibase.exception.DatabaseException: java.sql.SQLException: User not found: SA
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:393) ~[liquibase-core-3.5.1.jar!/:na]
	at org.airsonic.player.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:26) ~[classes!/:10.1.2-RELEASE]`
Full log: https://pastebin.com/KcnF84An

**Comment by lord-carlos** [comment]
A backup from a few month ago of the /db folder also works. Lost some data. But better then nothing.

**Comment by fxthomas** [comment]
How big is your `db` folder? Can you open it with a DB tool like [DBeaver](https://dbeaver.io/)?

**Comment by lord-carlos** [comment]
Mhh, the db folder is kinda empy'ish:
`du -sch db/*
0       db/airsonic.log
4.0K    db/airsonic.properties
4.0K    db/airsonic.script
8.0K    total`

The working restored one is larger:
`du -sch ../config/db/airsonic.*
4.0K    ../config/db/airsonic.lck
79M     ../config/db/airsonic.log
4.0K    ../config/db/airsonic.properties
6.4M    ../config/db/airsonic.script
85M     total`

I can open the .properties and .script files. But there is not much in it. 
Just: `CREATE SCHEMA PUBLIC AUTHORIZATION DBA
SET WRITE_DELAY 10`

It's just text documents. Or what file should I open with DBeaver?

**Comment by fxthomas** [comment]
You can add a HSQLDB connection to the `db/airsonic` database, like this in screenshot:

![Screenshot from 2019-03-09 00-20-49](https://user-images.githubusercontent.com/613594/54061449-38fcc180-4201-11e9-9d1a-5fa44af895f9.png)

...but your files look completely empty anyway, not sure what you can do in that case.

**Comment by shagr4th** [comment]
Same error with 10.3.1. The server is up and running for a few hours, crashes, then the db is corrupted.

**Comment by fxthomas** [comment]
@shagr4th I'm not entirely sure this is related since you're using a much newer version. When you say "crashes", what happens? Can you post logs?

**Comment by shagr4th** [comment]
`TEST`

**Comment by shagr4th** [comment]
I can't be sure root cause is related to Airsonic, since my server physically crashed (twice within a week) during the mediascannerservice step (I can't determine why, I only know there was unusual cpu/disk activity at that time). But on each time I had my db corrupted and these logs after a restart. To be fair, it looks like more of a hsqldb issue than an airsonic one

```2019-05-31 03:15:37.575  INFO --- org.airsonic.player.Application          : Starting Application v10.3.1-RELEASE on ac6a21acd394 with PID 7 (/airsonic/airsonic.war started by ? in /airsonic)
2019-05-31 03:15:37.609  INFO --- org.airsonic.player.Application          : The following profiles are active: legacy
2019-05-31 03:15:53.161  INFO --- o.a.p.service.SettingsService            : Java: 1.8.0_212, OS: Linux
2019-05-31 03:15:54.066  INFO --- org.airsonic.player.Application          : Detected Tomcat web server
2019-05-31 03:16:02.808 ERROR --- o.a.p.spring.SpringLiquibase             : ===============================================
2019-05-31 03:16:02.808 ERROR --- o.a.p.spring.SpringLiquibase             : An exception occurred during database migration
2019-05-31 03:16:02.808 ERROR --- o.a.p.spring.SpringLiquibase             : A rollback file has been generated at:q /airsonic/data/rollback.sql
2019-05-31 03:16:02.809 ERROR --- o.a.p.spring.SpringLiquibase             : Execute it within your database to rollback any changes
2019-05-31 03:16:02.820 ERROR --- o.a.p.spring.SpringLiquibase             : The exception is as follows

liquibase.exception.DatabaseException: java.sql.SQLException: User not found: SA
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:307) ~[liquibase-core-3.6.3.jar!/:na]
	at org.airsonic.player.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:26) ~[classes!/:10.3.1-RELEASE]```

**Comment by fxthomas** [comment]
If your server actually crashed (as in, kernel panic/freeze or hard power off) there's a good chance the HSQLDB files got corrupted. The version we use (1.8) is old and apparently not that reliable if the underlying hardware has issues.

_(FYI I personally keep daily snapshots, although I haven't had anything to report in a while.)_

**Comment by jvoisin** [comment]
We should really update HSQLDB :/

**Comment by ivakam** [comment]
So is there any known way to solve this issue? I'm dealing with the same problem myself where Airsonic just plain won't start at all. I guess I'd be ok with having to rebuild the db since it's better than the server not running at all.

**Comment by eharris** [comment]
It should be noted that there is a PR to upgrade HSQLDB in #876.

**Comment by eharris** [comment]
There's a newer version of the PR to upgrade HSQLDB in #1345.

**Comment by vadiof** [comment]
I am experiencing the same issue on 10.6.2 standalone version. This happened after update from 10.6.1 to 10.6.2. That time home dir for Airsonic was changed from /opt/airsonic to /var/airsonic and database was corrupted. Now it broken after reboot of my server. Reading similar issues' discussions I do not understand if it is possible for me to restore database? Any help?
